include/zephyr/dt-bindings/clock/numaker_m46x_clock.h

```diff
@@ -0,0 +1,2004 @@
+#define CLK_PLLCTL_180MHz_HXT                                                                      \
+	(CLK_PLLCTL_PLLSRC_HXT | CLK_PLLCTL_NR(3UL) | CLK_PLLCTL_NF(45UL) | CLK_PLLCTL_NO_2)
+#define CLK_PLLCTL_192MHz_HXT                                                                      \
+	(CLK_PLLCTL_PLLSRC_HXT | CLK_PLLCTL_NR(2UL) | CLK_PLLCTL_NF(32UL) | CLK_PLLCTL_NO_2)
+#define CLK_PLLCTL_200MHz_HXT                                                                      \
+	(CLK_PLLCTL_PLLSRC_HXT | CLK_PLLCTL_NR(3UL) | CLK_PLLCTL_NF(25UL) | CLK_PLLCTL_NO_1)
+
+#define CLK_PLLCTL_72MHz_HIRC                                                                      \
+	(CLK_PLLCTL_PLLSRC_HIRC | CLK_PLLCTL_NR(3UL) | CLK_PLLCTL_NF(36UL) | CLK_PLLCTL_NO_4)
+#define CLK_PLLCTL_80MHz_HIRC                                                                      \
+	(CLK_PLLCTL_PLLSRC_HIRC | CLK_PLLCTL_NR(3UL) | CLK_PLLCTL_NF(40UL) | CLK_PLLCTL_NO_4)
+#define CLK_PLLCTL_144MHz_HIRC                                                                     \
+	(CLK_PLLCTL_PLLSRC_HIRC | CLK_PLLCTL_NR(2UL) | CLK_PLLCTL_NF(24UL) | CLK_PLLCTL_NO_2)
+#define CLK_PLLCTL_160MHz_HIRC                                                                     \
+	(CLK_PLLCTL_PLLSRC_HIRC | CLK_PLLCTL_NR(3UL) | CLK_PLLCTL_NF(40UL) | CLK_PLLCTL_NO_2)
+#define CLK_PLLCTL_180MHz_HIRC                                                                     \
+	(CLK_PLLCTL_PLLSRC_HIRC | CLK_PLLCTL_NR(3UL) | CLK_PLLCTL_NF(45UL) | CLK_PLLCTL_NO_2)
+#define CLK_PLLCTL_192MHz_HIRC                                                                     \
+	(CLK_PLLCTL_PLLSRC_HIRC | CLK_PLLCTL_NR(2UL) | CLK_PLLCTL_NF(32UL) | CLK_PLLCTL_NO_2)
+#define CLK_PLLCTL_200MHz_HIRC                                                                     \
+	(CLK_PLLCTL_PLLSRC_HIRC | CLK_PLLCTL_NR(3UL) | CLK_PLLCTL_NF(25UL) | CLK_PLLCTL_NO_1)
+#define CLK_PLLCTL_384MHz_HIRC                                                                     \
+	(CLK_PLLCTL_PLLSRC_HIRC | CLK_PLLCTL_NR(1UL) | CLK_PLLCTL_NF(32UL) | CLK_PLLCTL_NO_2)
+
+
+/*  PLLFNCTL constant definitions. */
+
+#define CLK_PLLFNCTL1_PLLSRC_HXT  0x00000000UL
+#define CLK_PLLFNCTL1_PLLSRC_HIRC CLK_PLLFNCTL1_PLLSRC_Msk
+
+
+/*  MODULE constant definitions. */
+
+/*
+ * APBCLK(31:29)|CLKSEL(28:26)|CLKSEL_Msk(25:22)|CLKSEL_Pos(21:17)|CLKDIV(16:14)|
+ * CLKDIV_Msk(13:10)|CLKDIV_Pos(9:5)|IP_EN_Pos(4:0)
+ */
+
+#define MODULE_APBCLK(x)     (((x) >> 29) & 0x07UL)
+#define MODULE_CLKSEL(x)     (((x) >> 26) & 0x07UL)
+#define MODULE_CLKSEL_Msk(x) (((x) >> 22) & 0x0fUL)
+#define MODULE_CLKSEL_Pos(x) (((x) >> 17) & 0x1fUL)
+#define MODULE_CLKDIV(x)     (((x) >> 14) & 0x07UL)
+#define MODULE_CLKDIV_Msk(x) (((x) >> 10) & 0x0fUL)
+#define MODULE_CLKDIV_Pos(x) (((x) >> 5) & 0x1fUL)
+#define MODULE_IP_EN_Pos(x)  (((x) >> 0) & 0x1fUL)
+#define MODULE_NoMsk	     0x0UL
+#define NA		     MODULE_NoMsk
```

Comment:
this file includes > 2k lines, it's not clear to me what is expected to be used in DT. Things like `#define NA` or the lack of proper namespacing look suspicious. Can you please clarify? How is one expected to understand/use this file?

Reply:
In DT, it need the definition of clk module, clk select or clk divider, this file include all the options. This chip is for general purpose, so there are many options for each HW IP. Normally, developer will refer to existing property in dtsi  or check TRM doc for the config, just like what developer did on our proprietary BSP.

Reply:
Hi @gmarull,
Refined numaker_m46x_clock.h, to keep min. set of the definition of clk-module, clk-select & clk-divider for each H/W IP's clock config.
For ex. `<UART0_MODULE CLK_CLKSEL1_UART0SEL_HIRC CLK_CLKDIV0_UART0(1)>`:

-  UART0_MODULE: clk-module of UART0
-  CLK_CLKSEL1_UART0SEL_HIRC: choose HIRC as UART0's clk source
-  CLK_CLKDIV0_UART0(1): set clk divider for UART0

Please see [00f06a7](https://github.com/zephyrproject-rtos/zephyr/pull/55406/commits/00f06a740da7e685609dfd723e640a699aace213).